auth_oidc/controllers/main.py

```diff
@@ -48,3 +52,36 @@ def list_providers(self):
                     provider["auth_endpoint"], url_encode(params)
                 )
         return providers
+
+
+class OpenIDLogout(Session):
+    @http.route("/web/session/logout", type="http", auth="none")
+    def logout(self, redirect="/web/login"):
+        user = request.env["res.users"].sudo().browse(request.session.uid)
+        if user.oauth_provider_id.id:
+            p = (
+                request.env["auth.oauth.provider"]
+                .sudo()
+                .browse(user.oauth_provider_id.id)
+            )
+            if p.logout_endpoint:
+                r = redirect
+                if not r.startswith("http:") and not r.startswith("https:"):
+                    r = (
+                        request.env["ir.config_parameter"]
+                        .sudo()
+                        .get_param("web.base.url")
+                        + r
+                    )
+                logout_base_url = p.logout_endpoint
+                params = {}
+                if "?" in p.logout_endpoint:
+                    url_parts = p.logout_endpoint.split("?")
+                    logout_base_url = url_parts[0]
+                    params = url_decode(url_parts[1])
+                params["client_id"] = p.client_id
+                params["post_logout_redirect_uri"] = r
+                logout_url = f"{logout_base_url}?{url_encode(params)}"
```

Comment:
Is there documentation on how to run the test suite?

Reply:
@wluyima This [part of the odoo documentation](https://www.odoo.com/documentation/17.0/developer/reference/backend/testing.html?highlight=test%20suite#running-tests) should get you started. Let me know if you hit any issue.

Reply:
Thanks @sbidoul is there a way I can find documentation to run Odoo 14.0 locally from source? I was following the [15.0 documentation](https://www.odoo.com/documentation/15.0/administration/on_premise/source.html) and I can't seem to install the packages in the requirements.txt file, I get [this error](https://dpaste.com/8A7SVGMGA) when I try to install the packages, do you have any clues? I'm using Python **3.12.4**. I'm not sure if is not compatible, is there documentation on Odoo and Python versions compatibility?

Reply:
For Odoo 14 I generally use python 3.8. It will for sure not work with 3.12.

I recommend installing the Odoo dependencies in a virtual environment:

```
python3.8 -m venv ./venv
source ./venv/bin:activate
pip install -r https://raw.githubusercontent.com/odoo/odoo/14.0/requirements.txt
```

But if you don't use Odoo 14, why do you do this PR to the 14.0 branch?

Reply:
Thanks @sbidoul! I did say we're using Odoo 14 which is why this PR is issued against the 14.0 branch and I'm installing the dependencies in a virtual environment but I was using python 3.12, I'll try with python 3.8 and see if it works. I was actually able to later find the documentation for Odoo 14 [here](https://www.odoo.com/documentation/14.0/administration/install/source.html).